internal/provider/resources/block_test.go

```diff
@@ -1,39 +1,172 @@
 package resources_test
 
-// func fixtureAccBlockCreate(name string) string {
-// 	return fmt.Sprintf(`
-// 		data "prefect_workspace" "evergreen" {
-// 			handle = "github-ci-tests"
-// 		}
-// 		resource "prefect_block" "block" {
-// 			name = "%s"
-// 			type_slug = "secret"
-// 			workspace_id = data.prefect_workspace.evergreen.id
-// 			data = jsonencode({
-// 				"value": "hello, world"
-// 			})
-// 		}
-// 	`, name)
-// }
-
-// //nolint:paralleltest // we use the resource.ParallelTest helper instead
-// func TestAccResource_block(t *testing.T) {
-// 	resourceName := "prefect_block.block"
-// 	randomName := testutils.TestAccPrefix + acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum)
-
-// 	resource.ParallelTest(t, resource.TestCase{
-// 		ProtoV6ProviderFactories: testutils.TestAccProtoV6ProviderFactories,
-// 		PreCheck:                 func() { testutils.AccTestPreCheck(t) },
-// 		Steps: []resource.TestStep{
-// 			{
-// 				// Check creation + existence of the block resource
-// 				Config: fixtureAccBlockCreate(randomName),
-// 				Check: resource.ComposeAggregateTestCheckFunc(
-// 					resource.TestCheckResourceAttr(resourceName, "name", randomName),
-// 					resource.TestCheckResourceAttr(resourceName, "type_slug", "secret"),
-// 					resource.TestCheckResourceAttr(resourceName, "data", `{"value":"hello, world"}`),
-// 				),
-// 			},
-// 		},
-// 	})
-// }
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-testing/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/prefecthq/terraform-provider-prefect/internal/api"
+	"github.com/prefecthq/terraform-provider-prefect/internal/testutils"
+)
+
+func fixtureAccBlock(name string, value string) string {
+	return fmt.Sprintf(`
+resource "prefect_workspace" "workspace" {
+	name = "%s"
+	handle = "%s"
+}
+resource "prefect_block" "block" {
+	name = "%s"
+	type_slug = "secret"
+	workspace_id = prefect_workspace.workspace.id
+	data = jsonencode({
+		"value": "%s"
+	})
+	depends_on = [prefect_workspace.workspace]
+}
```

Comment:
before, i was using a dedicated, reserved workspace for tests + it would always exist. but i think creating one and tearing it down when testing resources that are workspace scoped might be a good way to run through full dependency integrations and ensure some deterministic behavior

Reply:
however, this is where i discovered some of the flakey async behavior that led me to add some retries

Reply:
Nice, ephemeral workspaces sounds like a good iteration